### ISSUE TITLE
Compilation error with @contract on function head with default argument

### ISSUE THREAD

**Issue by baldwindavid** [keathley/norm#73]
I've been adding `@contract` annotations to my functions throughout my codebase, but am running into a bit of a wall on handling the contract for a function head with a default argument.

```elixir
@contract create_user(permission :: is_atom(), attrs :: is_map()) ::
            one_of([
              {:ok, User.s()},
              {:error, is_changeset()}
            ])
def create_user(permission, attrs \\ %{})

def create_user(:super_administrate, attrs) do
  %User{}
  |> User.changeset(attrs)
  |> Repo.insert()
end

def create_user(:manage, attrs) do
  %User{}
  |> User.manage_create_changeset(attrs)
  |> Repo.insert()
end
```

I've tried multiple different ways, but it doesn't seem to want to compile when `def create_user(permission, attrs \\ %{})` is present.

I receive the following compilation error:

```
users.ex:60: implementation not provided for predefined def create_user/2
```

Is there a way to use `@contract` in this scenario?

**Comment by baldwindavid** [comment]
I could certainly change to providing the default via a regular function like below, but would prefer bodyless if it is possible.

```elixir
def create_user(permission) do
  create_user(permission, %{})
end
```

**Comment by baldwindavid** [comment]
Appears to be closed by #38
